chain/transaction.go

```diff
@@ -240,14 +249,19 @@ func (t *Transaction) PreExecute(
 	if err := t.Base.Execute(r.ChainID(), r, timestamp); err != nil {
 		return err
 	}
-	start, end := t.Action.ValidRange(r)
-	if start >= 0 && timestamp < start {
-		return ErrActionNotActivated
+	if len(t.Actions) > int(r.GetMaxActionsPerTx()) {
+		return ErrTooManyActions
 	}
-	if end >= 0 && timestamp > end {
-		return ErrActionNotActivated
+	for _, action := range t.Actions {
+		start, end := action.ValidRange(r)
+		if start >= 0 && timestamp < start {
+			return ErrActionNotActivated
```

Comment:
nit: it would probably be a lot more helpful to turn these into `fmt.Errorf` with info on which action or which index?